sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/block/CompactBlockDbStore.kt

```diff
@@ -5,58 +5,62 @@ import androidx.room.Room
 import androidx.room.RoomDatabase
 import cash.z.ecc.android.sdk.db.entity.CompactBlockEntity
 import cash.z.ecc.android.sdk.internal.SdkDispatchers
+import cash.z.ecc.android.sdk.internal.SdkExecutors
 import cash.z.ecc.android.sdk.internal.db.CompactBlockDao
 import cash.z.ecc.android.sdk.internal.db.CompactBlockDb
 import cash.z.wallet.sdk.rpc.CompactFormats
 import kotlinx.coroutines.withContext
+import kotlin.math.max
 
 /**
  * An implementation of CompactBlockStore that persists information to a database in the given
  * path. This represents the "cache db" or local cache of compact blocks waiting to be scanned.
- *
- * @param appContext the application context. This is used for creating the database.
- * @property dbPath the absolute path to the database.
  */
-class CompactBlockDbStore(
-    appContext: Context,
-    val dbPath: String
-) : CompactBlockStore {
-
-    private val cacheDao: CompactBlockDao
+class CompactBlockDbStore private constructor(
+    private val cacheDao: CompactBlockDao,
     private val cacheDb: CompactBlockDb
```

Comment:
why would you need a cacheDao in the constructor if it can be constructed from the cacheDb? is it for testing purposes?

Would it be possible to mistakenly pass a Dao and a CompactBlockDb that don't use the same database? What would happen in that case?

can this be done in an extension that allows the user to inject the  interface?

Reply:
Nice suggestion.  I'll make a change that simplifies this.